force-app/main/default/classes/MonthlyIterable_TEST.cls

```diff
@@ -538,4 +538,139 @@ public with sharing class MonthlyIterable_TEST {
             'List of Dates returned from the iterator was not as expected'
         );
     }
+
+    @IsTest
+    private static void shouldIterateEveryFirstWednesdayOfTheMonthOverAGivenYearUntilGivenEndDate() {
+        DateTime dateTimeStart = DateTime.newInstance(2020, 12, 1, 13, 0, 0);
```

Comment:
I wonder if we want to have a variety of different start dates? Test different months for start dates?

Reply:
Sure. I will add more tests